azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/AzMigrate_UpdateMachineProperties.ps1

```diff
@@ -85,6 +85,58 @@ Function ProcessItemImpl($processor, $csvItem, $reportItem) {
     }    
     #End Code for Target Subscription
 
+    $targetDiskName = $csvItem.UPDATED_TARGET_DISK_NAME
+    if ([string]::IsNullOrEmpty($targetDiskName)) {
+        $processor.Logger.LogTrace("UPDATED_TARGET_DISK_NAME is not mentioned for: '$($sourceMachineName)'")
+        $reportItem.AdditionalInformation = "UPDATED_TARGET_DISK_NAME is not mentioned for: '$($sourceMachineName)'"
+    }
+    else {
+        $params.Add("TargetDiskName", $targetDiskName)
+    }
+
+    $diskMapping = @()
+    
+    $paramsDisk1 = @{}
+    $osDiskId = $csvItem.OS_DISK_ID
+    $osDiskName = $csvItem.UPDATED_TARGET_OS_DISK_NAME
+    if ([string]::IsNullOrEmpty($osDiskId)) {
+        $processor.Logger.LogTrace("UPDATED_TARGET_OS_DISK_ID or UPDATED_TARGET_OS_DISK_NAME is not mentioned for: '$($sourceMachineName)', so OS Disk Name cannot be updated")
+        $reportItem.AdditionalInformation = "UPDATED_TARGET_OS_DISK_ID or UPDATED_TARGET_OS_DISK_NAME is not mentioned for: '$($sourceMachineName)', so OS Disk Name cannot be updated"
+    }
+    else {
+        $paramsDisk1.Add('DiskId', $osDiskId)
+        $paramsDisk1.Add('TargetDiskName', $osDiskName)
```

Comment:
Add $paramsDisk1.Add('IsOSDisk', true) for OS Disk.
As there is no difference in logic for OS Disk and Data disk except input name in CSV.
If data disk ID is passed in OS_DISK_ID, it will change it to OS disk as part of OS Disk Swap feature.